File: URI/Python/bemvindosebemvindasaoinverno.py
temp = [int(x) for x in input().split()]
temp1 = temp[0]
temp2 = temp[1]
temp3 = temp[2]

if (temp1 > temp2 and temp2 <= temp3):
    print(":)")

elif (temp1 < temp2 and temp2 >= temp3): 
    print(":(")

elif (temp1 < temp2 and temp2 < temp3 and (temp3-temp2)<(temp2-temp1)):
    print(":(")

elif (temp1 < temp2 and temp2 < temp3 and (temp3-temp2)>=(temp2-temp1)):
    print(":)")

elif (temp1 > temp2 and temp2 > temp3 and (temp2-temp3)<(temp1-temp2)):
    print(":)")

elif (temp1 > temp2 and temp2 > temp3 and (temp2-temp3)>=(temp1-temp2)):
    print(":(")

elif (temp1 == temp2):
    if(temp2 < temp3):
        print(":)")
    else: 
        print(":(")

File: URI/C/BemVindosEBemVindasAoInverno.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
typedef unsigned int uint;
typedef unsigned long long int ulli;

int main(void){
    int temp1, temp2, temp3;
    scanf("%d %d %d", &temp1, &temp2, &temp3);
    if (temp1 > temp2 && temp2 <= temp3) printf(":)\n");
    
    else if (temp1 < temp2 && temp2 >= temp3) printf(":(\n");
    
    else if (temp1 < temp2 && temp2 < temp3 && (temp3-temp2)<(temp2-temp1)) printf(":(\n");
    
    else if (temp1 < temp2 && temp2 < temp3 && (temp3-temp2)>=(temp2-temp1)) printf(":)\n");
    
    else if (temp1 > temp2 && temp2 > temp3 && (temp2-temp3)<(temp1-temp2)) printf(":)\n");
    
    else if (temp1 > temp2 && temp2 > temp3 && (temp2-temp3)>=(temp1-temp2)) printf(":(\n");
    
    else if (temp1 == temp2){
        if(temp2 < temp3) printf(":)\n");
        else printf(":(\n");
    }
    return 0;
}